testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/keys/ImportedRsaKeyProviderTest.java

```diff
@@ -69,13 +70,22 @@ public void addTestRealms(List<RealmRepresentation> testRealms) {
     }
 
     @Test
-    public void privateKeyOnly() throws Exception {
+    public void privateKeyOnlyForSig() throws Exception {
+        privateKeyOnly(ImportedRsaKeyProviderFactory.ID, KeyUse.SIG);
+    }
+
+    @Test
+    public void privateKeyOnlyForEnc() throws Exception {
+        privateKeyOnly(ImportedRsaEncKeyProviderFactory.ID, KeyUse.ENC);
```

Comment:
I'm afraid this test method does not check what we want to confirm.
In `private void privateKeyOnly(String providerId, KeyUse keyUse)`,
```
        assertEquals(kid, keys.getActive().get(Algorithm.RS256));
```
It is appropriate for an imported rsa key provider, but it is not appropriate for an imported rsa enc key provider.

For example, the following might be expected.
```
        assertEquals(kid, keys.getActive().get(Algorithm.RSA-OAEP));
```

However, as mentioned on other review comment, AbstractRsaKeyProvider's constructor set RS256 if a component model does not include an algorithm explicitly.
```
    public AbstractRsaKeyProvider(RealmModel realm, ComponentModel model) {
        this.model = model;
        this.status = KeyStatus.from(model.get(Attributes.ACTIVE_KEY, true), model.get(Attributes.ENABLED_KEY, true));
        this.algorithm = model.get(Attributes.ALGORITHM_KEY, Algorithm.RS256);
```
which means that an algorithm of an imported rsa enc key provider without setting its algorithm explicitly becomes RS256. It is not appropriate.

Therefore, how about modifying AbstractRsaKeyProvider's constructor as follows?
```
    public AbstractRsaKeyProvider(RealmModel realm, ComponentModel model) {
        this.model = model;
        this.status = KeyStatus.from(model.get(Attributes.ACTIVE_KEY, true), model.get(Attributes.ENABLED_KEY, true));
        String keyUse = Optional.ofNullable(model.get(Attributes.KEY_USE)).orElse(KeyUse.SIG.name());
        if (keyUse.equals(KeyUse.ENC.name())) {
            this.algorithm = model.get(Attributes.ALGORITHM_KEY, JWEConstants.RSA_OAEP);
        } else {
            this.algorithm = model.get(Attributes.ALGORITHM_KEY, Algorithm.RS256);
        }
...
```

Reply:
I checked the test values in debug.
You're right, this test is not what we want.
I'll fix this.